packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -374,7 +375,11 @@ export default class {
     if (resourceSetId != null) {
       await this.allocateLimitsInResourceSet(resourcesUsage, resourceSetId)
     }
-    if (previousResourceSetId !== undefined) {
+
+    const previousResourceSetExists = find(await this.getAllResourceSets(), {
+      id: previousResourceSetId,
+    })
```

Comment:
We should not fetch all resource sets and then find if the one we are looking for exists.

`await this._store.has(id)`.

Reply:
And it should be done only if `previousResourceSetId !== undefined`.